src/stubs/sm_exit.s

```diff
@@ -80,6 +80,14 @@ __reti_entry:
     .global __ret_entry
     .type __ret_entry,@function
 __ret_entry:
+    ; If we return from an OCALL, overwrite the stored caller_id
```

Comment:
hm, not sure, but I think this may be insecure(!)

The atomic entry perion `sm_entry.S` already finished here so doing a callerID instruction does *not* guarantee this is still the prev SM in the thread right?

Why don't we reserve a variable to store the callerID in the atomic entry and then we can move it optionally into the SSA later?

Reply:
Good catch. 
Alternatively we move the eint to after the branch and add an eint in the ret entry after the pop r15 line. I am not sure what is arguably more overhead, adding a variable for this or extending the atomic section by some few cycles? I would vote for extending the atomic section.

Reply:
no, I vote for keep the atomic section as short as possible :)

and also it's much cleaner and easier to understand when the atomic section does not cross files I find, a bit unfortunate that's already not the case for sm_reti_entry and sm_isr though

Reply:
okay I think I fixed it as per 7b51d8e

@fritzalder  pls double-check (there was also an incorrect # instead of & I think) and mark resolved when done ^^

Reply:
to be secure, the sm_entry stub should still verify that there's actually an open entry point when accepting a call with 0xffff

this is described in issue #13 

let's discuss this next week

Reply:
This part looks good and I'll resolve it. Further discussion on #13 will be done there to not clutter up this PR discussion even more :D